drivers/phoenixcontact_modbus.c

```diff
@@ -49,21 +54,58 @@ void upsdrv_initinfo(void)
 	upsdebugx(2, "upsdrv_initinfo");
 
 	dstate_setinfo("device.mfr", "Phoenix Contact");
-	dstate_setinfo("device.model", "QUINT-UPS/24DC");
-
+	
 	/* upsh.instcmd = instcmd; */
 	/* upsh.setvar = setvar; */
+
+	uint16_t FWVersion;
+
+	mrir(modbus_ctx, 0x0004, 1, &FWVersion);
```

Comment:
Makes sense to also populate `ups.firmware`

Reply:
I agree, not sure if the variable should be defined somewhere else or i can just call the dstate_setinfo function.

Reply:
Well, the global variable makes sense if this info would be used elsewhere (e.g. instead of those `OLD_UPS/NEW_UPS` defines?)